### DISCUSSION TITLE
Closing ticket by Ticket Owner Only

### DISCUSSION THREAD

**Discussion by LeeTV1** [cerb/cerb-release#63] [Q&A] [answered]
Hello
Where do I begin by adding code (and what code do I add) only to allow a ticket owner to close a ticket? 

(i.e., if another user, a non-ticket-owner, opens another user's ticket and they try to close it, how can we prevent the closure of the ticket if that user is not the ticket owner? (i.e., prevent the closure, warn the user, and log the attempt to close).

TIA.
Lee

**Comment by jstanden** [comment, marked as answer]
Unfortunately, it's currently easier to revert the change after the fact than to prevent it. That should change in the near future with record update validator automations.

You can use the [record.changed](https://cerb.ai/docs/automations/events/record.changed/) automation event to detect when the `status` of a ticket record transitions into the `closed` status. It includes the actor (e.g. `actor__type` == `worker`, `actor_id` == `123`).

You can compare `actor_id` to `record_owner_id`. The event also includes `was_record_owner_id` to revert.

You can trigger any kind of action when it happens (log entry, comment, Slack post, email, webhook, etc).

It's also possible to cut down on the opportunity for someone to close a ticket improperly. On ticket profiles, you can edit the 'Status' widget and add a `hidden@bool:` condition to the 'Close' shortcut button. If a worker isn't the current ticket owner you can hide the button entirely. It would still be visible in worklists or the popup record editor.

We're planning to get record editor validation automations into 10.5. That would let you use custom logic to accept/deny any record changes.

**Comment by LeeTV1** [comment]
I still have this email on my list of things to follow-up. I’m guessing that this will be easier in 10.5?

Lee

**Reply by jstanden** [reply]
Hi Lee!

Yeah, at the moment you're better off catching the change and reverting it with a `record.changed` automation.

We have several feature requests that would benefit from having custom pre-save validation on record editors. It's high on my list for 10.5.
